scripts/release/mule/sign/sign.sh

```diff
@@ -21,15 +21,23 @@ SIGNING_KEY_ADDR=dev@algorand.com
 OS_TYPE=$(./scripts/release/mule/common/ostype.sh)
 ARCHS=(amd64 arm arm64)
 ARCH_BITS=(x86_64 armv7l aarch64)
-
-# It seems that copying/mounting the gpg dir from another machine can result in insecure
-# access privileges, so set the correct permissions to avoid the following warning:
-#
-#   gpg: WARNING: unsafe permissions on homedir '/root/.gnupg'
+# Note that we don't want to use $GNUPGHOME here because that is a documented env var for the gnupg
+# project and if it's set in the environment mule will automatically pick it up, which could have
+# unintended consequences and be hard to debug.
 #
-find /root/.gnupg -type d -exec chmod 700 {} \;
-find /root/.gnupg -type f -exec chmod 600 {} \;
+# By naming it something other than $GNUPGHOME, it's essentially acting as an opt-in.
+GPG_DIR=${GPG_DIR:-/root/.gnupg}
```

Comment:
Good idea, especially if we eventually stop running containers as root, as we should. It does mean we never run this when running scripts _locally_, but that's probably fine.